### PR TITLE
Fix issue with "bad" keyword in object and function definition

### PR DESCRIPTION
**Pull request by Bob-Thomas**
#553

### CODE REVIEW

**Comment by xdjinnx** [comment]
Is it possible to add a test for this?

**Comment by Bob-Thomas** [comment]
@xdjinnx Working on it

**Comment by Bob-Thomas** [comment]
@xdjinnx Added a tiny test for keyword use inside an object.

File: test/avoid-scope-typos.js [hunk 1]
```diff
@@ -43,6 +47,20 @@ valid.push({
...
 
+/** Test the usage of the keyword inside an object */
+bad.forEach(function(b) {
+    valid.push({
+        code: '$ctrl.test = {' + b + ':\'test\'}'
+    });
+});
+
+/** Test for using none reserved keywords as a function name */
+nonReservedBad.forEach(function(b) {
```
Comment (Bob-Thomas) [inline test/avoid-scope-typos.js]: test for #507

**Comment by Bob-Thomas** [comment]
@xdjinnx any reason the travis is failing ?

**Comment by Bob-Thomas** [comment]
@Gillespie59 @xdjinnx

**Comment by Gillespie59** [comment]
Thanks @Bob-Thomas

**Pull request merged**